src/js/reducers/layoutReducer.ts

```diff
@@ -0,0 +1,64 @@
+import {Map} from 'immutable';
+import {ActionType, getType} from 'typesafe-actions';
+import {LayoutState} from '../store/factory/Layout';
+import * as layoutActions from '../actions/layoutActions';
+
+/**
+ * This reducer handles layout updates
+ * @param {Object} state - An Immutable state object
+ * @param {Object} action - An action object
+ */
+export function layoutReducer(state: LayoutState,
+  action: ActionType<typeof layoutActions>): LayoutState {
+  // handle layout actions
+  const id = action.meta;
```

Comment:
consider doing `const id = String(action.meta);` since you cast it to string everywhere below